src/machine.rs

```diff
@@ -182,6 +215,9 @@ pub struct Evaluator<'tcx> {
 
     /// The "time anchor" for this machine's monotone clock (for `Instant` simulation).
     pub(crate) time_anchor: Instant,
+
+    /// Cached `TyLayout`s for primitive data types that are commonly used inside Miri.
+    primitive_layouts: PrimitiveLayouts<'tcx>,
```

Comment:
Could you add a FIXME to go search through the Miri codebase for `layout_of` calls that could have their cache added here?